Add voting info to labels selector

kevinearldenny opened the pull request:
## Overview

Adds voting info to labels selector, which display vote totals for geounits when selected.

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

### Demo

![image](https://user-images.githubusercontent.com/66973361/123856322-f3db8c80-d8ee-11eb-9bf2-a3cfe21dfb2c.png)


### Notes

We may want to handle the formatting of the field names here - I wasn't able to devise an easy solution for this, and I guess the labels aren't _terrible_ currently (Democrat20, Other party20). I was thinking that it might make sense to change these labels further upstream in the geojson processing, but that likely requires some input from other stakeholders first.

## Testing Instructions

- Go to a state with voting data and look in the labels dropdown - expect to see voting fields displayed
- Click a voting field, expect to see voting totals displayed

Closes #731

maurizi commented on src/client/components/MapHeader.tsx:
```diff
@@ -145,8 +145,14 @@ const MapHeader = ({
     : undefined;
-  const labelOptions = metadata
-    ? metadata.demographics.map(val => (
+  const labelFields =
+    metadata && metadata.demographics
+      ? metadata.voting
+        ? [...metadata.demographics, ...metadata.voting]
+        : [...metadata.demographics]
+      : undefined;
+  const labelOptions = labelFields
+    ? labelFields.map(val => (
         <option key={val.id} value={val.id}>
           {capitalizeFirstLetter(val.id)}
```
```suggestion
  const labelFields =
    metadata && metadata.demographics
      ? metadata.voting
        ? [
           ...metadata.demographics.map(file => { id: file.id, label: file.id }),
           ...metadata.voting.map(file => {
             id: file.id,
             // Fix suffix on voting IDs if needed
             label: file.id.endsWith("16") || file.id.endsWith("20") ? file.id.slice(0, -2) + " '" + file.id.slice(-2) : file.id
           })]
        : [...metadata.demographics.map(file => { id: file.id, label: file.id })]
      : undefined;
  const labelOptions = labelFields
    ? labelFields.map(val => (
        <option key={val.id} value={val.id}>
          {capitalizeFirstLetter(val.label)}
```

maurizi reviewed (changes requested):
>  I was thinking that it might make sense to change these labels further upstream in the geojson processing, but that likely requires some input from other stakeholders first.

We've made quite a few code changes that make use of the `16` / `20` suffix, so I don't think changing these upstream now would a good use of ours or Brian's time.

Left a (rough & untested) suggestion of how we might coax these IDs into appropriate labels

kevinearldenny commented:
Okay, updated labels are now implemented: 
![image](https://user-images.githubusercontent.com/66973361/124148911-2061e680-da5e-11eb-8702-bb7e30e1a4db.png)

pcaisse reviewed (approved):
:tada:
